### ISSUE TITLE
torch.distributed launch.py is hanged.  (pid, sts) = os.waitpid(self.pid, wait_flags)

### ISSUE THREAD

**Issue by zyXu1996** [pytorch/pytorch#45042]
## To Reproduce

1.running in the terminal :python3.5 -m torch.distributed.launch --nproc_per_node=2 --master_addr 127.0.0.2 --master_port 29501 train.py
1.after many epochs training, it randomly hanged, The procedure still exists but hanged
1.This often happens on weight models, a light-weight model that only runs for several hours or one day usually don't have this bug.

File "/usr/lib/python3.5/runpy.py", line 184, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.5/dist-packages/torch/distributed/launch.py", line 235,
 in <module>
    main()
  File "/usr/local/lib/python3.5/dist-packages/torch/distributed/launch.py", line 228,
 in main
    process.wait()
  File "/usr/lib/python3.5/subprocess.py", line 1658, in wait
    (pid, sts) = self._try_wait(0)
  File "/usr/lib/python3.5/subprocess.py", line 1608, in _try_wait
    (pid, sts) = os.waitpid(self.pid, wait_flags)

## Expected behavior

The program should continue to run, but has been suspended, as if the child process died

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0):1.1
 - OS (e.g., Linux):ubuntu 16.04.6
 - How you installed PyTorch (`conda`, `pip`, source):pip
 - Build command you used (if compiling from source):
python3.5 -m torch.distributed.launch --nproc_per_node=2 --master_addr 127.0.0.2 --master_port 29501 train.py

 - Python version:3.6.7
 - CUDA/cuDNN version:CUDA Version 10.0.130  cudnn:7.5.0
 - GPU models and configuration:four rtx 2080ti
 - Any other relevant information:

**Comment by zyXu1996** [comment]
`the code setting like:
parser.add_argument("--local_rank", type=int, default=0)

 cudnn.benchmark = True
 cudnn.deterministic = False
 cudnn.enabled = True
 distributed = True
 device = torch.device(('cuda:{}').format(args2.local_rank))

    if distributed:
        torch.cuda.set_device(args2.local_rank)
        torch.distributed.init_process_group(
            backend="nccl", init_method="env://",
        )
    if distributed:
        train_sampler = DistributedSampler(potsdam_train)
    else:
        train_sampler = None
    dataloader_train = DataLoader(
        potsdam_train,
        batch_size=args2.train_batchsize,
        shuffle=True and train_sampler is None,
        num_workers=4,
        pin_memory=True,
        drop_last=True,
        sampler=train_sampler)

    model = nn.SyncBatchNorm.convert_sync_batchnorm(model)
    model = model.to(device)
    model = nn.parallel.DistributedDataParallel(
        model, device_ids=[args2.local_rank], output_device=args2.local_rank)`

**Comment by malfet** [comment]
Can you please try reproducing the same issue using PyTorch-1.6.0?

**Comment by zyXu1996** [comment]
> Can you please try reproducing the same issue using PyTorch-1.6.0?

OK, I will try it later

**Comment by zyXu1996** [comment]
> Can you please try reproducing the same issue using PyTorch-1.6.0?

Sorry for bothering you, I don't apply PyTorch-1.6.0, but tried others' setting, it works well:
`    torch.manual_seed(args.seed)
    torch.cuda.manual_seed(args.seed)
    random.seed(args.seed)
    np.random.seed(args.seed)
    torch.backends.cudnn.enabled = True
    torch.backends.cudnn.benchmark = True`
In my code, I don't fix random seed，the setting is:
`    cudnn.benchmark = True
    cudnn.deterministic = False
    cudnn.enabled = True
`
This is the only difference，may it leads to the bug?

**Comment by RicoSuaveGuapo** [comment]
I have a similar issue, is there any explanation for why the above happen?

**Comment by donand** [comment]
I'm having the same issue, the training hangs and the only solution is to restart it.
